src/cgame/cg_character.c

```diff
@@ -450,6 +450,17 @@ qboolean CG_RegisterCharacter(const char *characterFile, bg_character_t *charact
 		return qfalse;  // the parser will provide the error message
 	}
 
+	// override for demo backwards compat
+	if (cg.demoBackwardsCompat.characterAnimationGroup[0] != '\0') {
+		CG_Printf("DEMOBACKWARDSCOMPAT: Loading animationGroup '%s'\n", cg.demoBackwardsCompat.characterAnimationGroup);
+		strcpy(characterDef.animationGroup, cg.demoBackwardsCompat.characterAnimationGroup);
+	}
+
+	if (cg.demoBackwardsCompat.characterAnimationScript[0] != '\0') {
+		CG_Printf("DEMOBACKWARDSCOMPAT: Loading animationScript '%s'\n", cg.demoBackwardsCompat.characterAnimationScript);
+		strcpy(characterDef.animationScript, cg.demoBackwardsCompat.characterAnimationScript);
```

Comment:
There's a safe string copy function in `q_shared.c`, should use that here rather than raw `strcpy`

https://github.com/etlegacy/etlegacy/blob/af6404d1b335cf8160b7fdb93a6b5dce6a5d708a/src/qcommon/q_shared.c#L1576-L1594

```suggestion
		Q_strncpyz(characterDef.animationGroup, cg.demoBackwardsCompat.characterAnimationGroup, sizeof(characterDef.animationGroup));
	}

	if (cg.demoBackwardsCompat.characterAnimationScript[0] != '\0') {
		CG_Printf("DEMOBACKWARDSCOMPAT: Loading animationScript '%s'\n", cg.demoBackwardsCompat.characterAnimationScript);
		Q_strncpyz(characterDef.animationScript, cg.demoBackwardsCompat.characterAnimationScript, sizeof(characterDef.animationScript));
```